### ISSUE TITLE
论文疑惑，希望能得到您帮助

### ISSUE THREAD

**Issue by June-Li** [opendatalab/MinerU-Diffusion#16]
这是 mineru diffusion 论文里的图，能帮我解读下不，我主要是没理解右边那个矩阵图啥意思，label 和 mask 后的堆在一起形成一个矩阵

<img width="893" height="470" alt="Image" src="https://github.com/user-attachments/assets/506e0ae6-0799-4a04-806b-5bac1b1ddc3d" />

**Comment by jhdonghj** [comment]
右侧矩阵表示的是训练时的 block-wise attention mask。我们将输出序列划分为多个 block：同一个 block 内 token 之间可以双向 attention，用于并行预测被 mask 的位置；不同 block 之间保持 causal attention，即当前 block 只能看到之前的 block，不能看到未来的 block。

因此，这个 mask 的作用是同时支持 块内并行 diffusion 预测 和 块间自回归式顺序约束。

**Comment by June-Li** [comment]
ok，懂了，多谢多谢

**Comment by andyjiang1116** [comment]
> 右侧矩阵表示的是训练时的 block-wise attention mask。我们将输出序列划分为多个 block：同一个 block 内 token 之间可以双向 attention，用于并行预测被 mask 的位置；不同 block 之间保持 causal attention，即当前 block 只能看到之前的 block，不能看到未来的 block。
> 
> 因此，这个 mask 的作用是同时支持 块内并行 diffusion 预测 和 块间自回归式顺序约束。

@jhdonghj 您好，我这里还有个疑惑，那具体训练的时候输入序列是拼接的[vision+prompt+clean label + masked label]吗，还是只有[vision + prompt + masked label]呢？因为我看sdar模型里面训练是noisy+clean双序列的

**Comment by Niujunbo2002** [comment]
训练的时候输入序列是拼接的[vision+prompt+clean label + masked label]
我们与 yinjie wang 的 dllm-rl 仓库中的训练方式对齐.

 牛俊博
***@***.***



---- 回复的原邮件 ----
发件人 andyjiang ***@***.***> 发送日期 2026年07月15日 15:26 收件人 opendatalab/MinerU-Diffusion ***@***.***> 抄送人 Subscribed ***@***.***> 主题 Re: [opendatalab/MinerU-Diffusion] 论文疑惑，希望能得到您帮助 (Issue #16)  
andyjiang1116 left a comment (opendatalab/MinerU-Diffusion#16)
右侧矩阵表示的是训练时的 block-wise attention mask。我们将输出序列划分为多个 block：同一个 block 内 token 之间可以双向 attention，用于并行预测被 mask 的位置；不同 block 之间保持 causal attention，即当前 block 只能看到之前的 block，不能看到未来的 block。
因此，这个 mask 的作用是同时支持 块内并行 diffusion 预测 和 块间自回归式顺序约束。
您好，我这里还有个疑惑，那具体训练的时候输入序列是拼接的[vision+prompt+clean label + masked label]吗，还是只有[vision + prompt + masked label]呢？因为我看sdar模型里面训练是noisy+clean双序列的
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications, keep track of coding agent tasks and review pull requests on the go with GitHub Mobile for iOS and Android. Download it today!

**Comment by andyjiang1116** [comment]
> 训练的时候输入序列是拼接的[vision+prompt+clean label + masked label]
> 我们与 yinjie wang 的 dllm-rl 仓库中的训练方式对齐.
> 
>  牛俊博
> ***@***.***
> […](#)

好的，感谢回复，顺便问下，咱们这边大概啥时候release training code 呢？

**Comment by Niujunbo2002** [comment]
暂时尚未有明确的计划，v2 迭代完成后可能会考虑 release.
 牛俊博
***@***.***




---- 回复的原邮件 ----
发件人 andyjiang ***@***.***> 发送日期 2026年07月15日 17:25 收件人 opendatalab/MinerU-Diffusion ***@***.***> 抄送人 Junbo Niu ***@***.***>,
Comment ***@***.***> 主题 Re: [opendatalab/MinerU-Diffusion] 论文疑惑，希望能得到您帮助 (Issue #16)  
andyjiang1116 left a comment (opendatalab/MinerU-Diffusion#16)
训练的时候输入序列是拼接的[vision+prompt+clean label + masked label]
 我们与 yinjie wang 的 dllm-rl 仓库中的训练方式对齐.
牛俊博
 @.***
 …
好的，感谢回复，顺便问下，咱们这边大概啥时候release training code 呢？
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications, keep track of coding agent tasks and review pull requests on the go with GitHub Mobile for iOS and Android. Download it today!

**Comment by andyjiang1116** [comment]
> 暂时尚未有明确的计划，v2 迭代完成后可能会考虑 release.
>  牛俊博
> ***@***.***
> […](#)

好的，期待～
